.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,45 @@
+name: "Code scanning - action"
+
+on:
+  push:
```

Comment:
Only thing I would suggest is to add filters so this doesn't run on pushes that only involve changes to markdown files. I had a recent PR in `setup-python` that only involved changes to the `README` and this felt a little pointless to have codeql running for that. This also takes 4+ minutes on average to run so the more selective we are, the better imo.
```
    paths-ignore:
      - '**.md'
```

Similar filters in some other [worfklows](https://github.com/actions/setup-python/blob/ee653cd47f3d328dcb2ae4ad7a0a9ab2c5b34804/.github/workflows/test.yml#L6)